src/Symfony/Component/Routing/Annotation/Route.php

```diff
@@ -106,6 +106,22 @@ public function getName()
 
     public function setRequirements($requirements)
     {
+        if (isset($requirements['_method'])) {
+            if (0 === count($this->methods)) {
+                $this->methods = explode('|', $requirements['_method']);
```

Comment:
I see. I was thinking about the use of the annotation class independtly, but that is not what we need to keep track of.